src/managers/config.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling configuration building + writing."""
+
+import logging
+
+from ops.model import ConfigData
+
+from core.cluster import ClusterState
+from core.workload import WorkloadBase
+from literals import CONFIG_PATH
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PROPERTIES = """
```

Comment:
I think we can make use of string templates with placeholders here. It would be more readable IMO. It would also avoid the string concatenation from lists.